store/sqlstore/channel_member_history_store.go

```diff
@@ -76,14 +109,37 @@ func (s SqlChannelMemberHistoryStore) GetUsersInChannelDuring(startTime int64, e
 			AND (cmh.LeaveTime IS NULL OR cmh.LeaveTime >= :StartTime)
 			ORDER BY cmh.JoinTime ASC`
 
-		params := map[string]interface{}{"ChannelId": channelId, "StartTime": startTime, "EndTime": endTime}
-		var histories []*model.ChannelMemberHistory
-		if _, err := s.GetReplica().Select(&histories, query, params); err != nil {
-			result.Err = model.NewAppError("SqlChannelMemberHistoryStore.GetUsersInChannelAt", "store.sql_channel_member_history.get_users_in_channel_during.app_error", params, err.Error(), http.StatusInternalServerError)
-		} else {
-			result.Data = histories
+	params := map[string]interface{}{"ChannelId": channelId, "StartTime": startTime, "EndTime": endTime}
+	var histories []*model.ChannelMemberHistory
+	if _, err := s.GetReplica().Select(&histories, query, params); err != nil {
+		return nil, err
+	} else {
+		return histories, nil
+	}
+}
+
+func (s SqlChannelMemberHistoryStore) getFromChannelMembersTable(startTime int64, endTime int64, channelId string) ([]*model.ChannelMemberHistory, error) {
+	query := `
+		SELECT DISTINCT
+  			ch.ChannelId,
+  			ch.UserId,
+  			u.email
+		FROM ChannelMembers AS ch
+		INNER JOIN Users AS u ON ch.UserId = u.id
+		WHERE ch.ChannelId = :ChannelId`
```

Comment:
This method is less accurate, because it assumes that a user might have been in the channel during the export period if they have ever been in that channel at any point in the past. It's the best that we can do, because ChannelMembers doesn't really track join and leave times, so this method will include people who weren't actually in the channel. Importantly though, it won't miss anybody - there won't be somebody who was actually in the channel that isn't mentioned in the export, which is crucial.